lib/datadog/tracing/distributed/datadog.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+# typed: true
+
+require_relative '../metadata/ext'
+require_relative '../trace_digest'
+require_relative 'datadog_tags_codec'
+
+module Datadog
+  module Tracing
+    module Distributed
+      # Datadog-style trace propagation.
+      class Datadog
+        def initialize(
+          trace_id: Ext::HTTP_HEADER_TRACE_ID,
+          parent_id: Ext::HTTP_HEADER_PARENT_ID,
+          sampling_priority: Ext::HTTP_HEADER_SAMPLING_PRIORITY,
+          origin: Ext::HTTP_HEADER_ORIGIN,
+          tags: Ext::HTTP_HEADER_TAGS,
```

Comment:
See my notes on the b3 class about calling these headers :)

Reply:
Done!